### PR TITLE
resource header dropdown

### PR DESCRIPTION
**Pull request by der-john**
this replaces #2530. Like the resource header (see #2511), it's now first implemented in the idea collection, and may be implemented elsewhere soon.

Icons are working, but I'm not entirely sure how I accomplished it.

### CODE REVIEW

**Comment by der-john** [comment]
sorry, 	f7c7db0 (make dropdown vanish when user clicks elsewhere) is still buggy.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceDropdown.html [hunk 1]
```diff
@@ -0,0 +1,96 @@
+<div class="dropdown resource-header-actions" data-ng-if="!data.modals.modal && data.isShowDropdownMenu">
```
Comment (der-john) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceDropdown.html]: `data.isShowDropdownMenu` will come into play in 9d8a475 (make dropdown button disappear when alerts appear). sorry for the confusion.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceDropdown.html [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    <a href="" tabindex="0" aria-haspopup="isShowDropdown" aria-expanded="isShowDropdown" data-ng-click="data.toggleDropdown()">
+        <i class="icon-three-dots dropdown-icon dropdown-header-icon"></i>
+    </a>
+    <div class="dropdown-menu dropdown-menu-right" data-ng-if="isShowDropdown" aria-labelledby="dropdownMenu1">
+        <adh-view-action
+            data-ng-if="data.edit && data.resourcePath && options.PUT"
+            data-view="edit"
+            data-label="TR__EDIT"
+            data-resource-path="{{data.resourcePath}}">
+	    <i class="icon-pencil dropdown-icon"></i>
```
Comment (der-john) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceDropdown.html]: indent will be fixed later.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/IdeaCollection/Proposal/Proposal.ts [hunk 3]
```diff
@@ -235,7 +235,8 @@ export var detailDirective = (
             path: "@",
             processOptions: "="
         },
-        link: (scope : IScope) => {
+        link: (scope) => {
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/IdeaCollection/Proposal/Proposal.ts]: Why did you remove the type?

Comment (der-john) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/IdeaCollection/Proposal/Proposal.ts]: Good question. I probably briefly put a new variable into the scope, but I don't remember.

File: src/adhocracy_frontend/adhocracy_frontend/static/stylesheets/scss/components/_resource_header.scss [hunk 4]
```diff
@@ -76,12 +83,6 @@ actions (edit, share, report, print, ...).
     }
 }
 
-.resource-header-actions {
-    position: absolute;
-    top: $padding-small;
-    right: $padding-small;
-}
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/stylesheets/scss/components/_resource_header.scss]: Why was this moved?

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts [hunk 5]
```diff
@@ -162,7 +242,9 @@ export var hideActionDirective = (
 ) => {
     return {
         restrict: "E",
-        template: "<a class=\"{{class}}\" href=\"\" data-ng-click=\"hide();\">{{ 'TR__HIDE' | translate }}</a>",
+        transclude: true,
+        template: "<a class=\"{{class}}\" href=\"\" data-ng-click=\"hide();\"><ng-transclude></ng-transclude>" +
+            "{{ 'TR__HIDE' | translate }}</a>",
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: Adding transclusion to all of these is overly complex IMHO. Why not add the icons to the template directly?

File: src/adhocracy_frontend/adhocracy_frontend/static/stylesheets/scss/components/_resource_header.scss [hunk 6]
```diff
@@ -34,6 +34,13 @@ actions (edit, share, report, print, ...).
     position: relative;
     padding: $padding-small;
     margin-bottom: $padding-small;
+    z-index: 0;
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/stylesheets/scss/components/_resource_header.scss]: 0 is the default value, so this should not be necessary.

File: src/adhocracy_frontend/adhocracy_frontend/static/stylesheets/scss/components/_resource_header.scss (hunk 4 above)
Comment (der-john) [inline src/adhocracy_frontend/adhocracy_frontend/static/stylesheets/scss/components/_resource_header.scss]: By accident. Needs to be in.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts (hunk 5 above)
Comment (der-john) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: Good question:
1. `modalActionDirective` and `viewActionDirective` use different icons for different actions. `ng-transclude` comes in handy to get those different icons into them.
2. I strove to make the different action directives as similar as possible.
3. En passant, `ng-transclude` includes a little whitespace between icon and text.

You're welcome to relinquish `ng-transclude` where possible (you compromise (2) and need to solve (3) then, though).

File: src/adhocracy_frontend/adhocracy_frontend/static/stylesheets/scss/components/_resource_header.scss (hunk 6 above)
Comment (der-john) [inline src/adhocracy_frontend/adhocracy_frontend/static/stylesheets/scss/components/_resource_header.scss]: ok. thanks.

File: src/adhocracy_frontend/adhocracy_frontend/static/stylesheets/scss/components/_dropdown.scss [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+        }
+
+        &:hover,
+        &:focus {
+            text-decoration: none;
+            color: $color-brand-two-normal;
+            background-color: $color-structure-normal;
+        }
+    }
+}
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/stylesheets/scss/components/_dropdown.scss]: The dropdown menu should be hidden on print

**Comment by xi** [comment]
There are still some open questions and tasks that need to be handled before this can be merged:

-   [ ] re-build icons for all projects
-   [ ] do not show dropdown button if the dropdown menu would be empty

**Comment by der-john** [comment]
Update: icons in euth and pcompass *should* work. I'm having testing problems. :(

**Comment by der-john** [comment]
As for the empty-dropdown issue, i.e.
>  do not show dropdown button if the dropdown menu would be empty

this is a (non-trivial) theoretical issue right now: dropdowns are only implemented in idea collections, and there, they are never empty. So we can merge.

**Comment by der-john** [comment]
416ffa9 adds unnecessary whitespace that gets underlined. Can we revert it?

**Comment by xi** [comment]
> 416ffa9 adds unnecessary whitespace that gets underlined. Can we revert it?

I cannot find any place where this is underlined.
